File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("change owed: ");
    }
    while (cents < 0);

    // calculate how many quarters you should give customer
    int quarters = calculate_quarters(cents);

    // subtract the value of those quarters from cents
    cents = cents - (quarters * 25);

    // calculate how many dimes you should give customer
    int dimes = calculate_dimes(cents);

    // subtract the value of those dimes from remaining cents
    cents = cents - (dimes * 10);

    // calculate how many nickels you should give customer
    int nickels = calculate_nickels(cents);

    // subtract the value of those nickels from remaining cents
    cents = cents - (nickels * 5);

    // calculate how many pennies you should give customer
    int pennies = calculate_pennies(cents);

    // subtract the value of those pennies from remaining cents
    cents = cents - (pennies * 1);

    // sum the number of quarters, dimes, nickels, and pennies used
    int total =
        quarters + dimes + nickels + pennies; // Declare and assign a value to total_coins
    printf("Total coins: %d\n", total);

    return 0;
}

int calculate_quarters(int cents)
{
    int i;
    for (i = 0; cents >= 25; i++)
    {
        cents = cents - 25;
    }
    return i;
}

int calculate_dimes(int cents)
{
    int i;
    for (i = 0; cents >= 10; i++)
    {
        cents = cents - 10;
    }
    return i;
}

int calculate_nickels(int cents)
{
    int i;
    for (i = 0; cents >= 5; i++)
    {
        cents = cents - 5;
    }
    return i;
}

int calculate_pennies(int cents)
{
    int i;
    for (i = 0; cents >= 1; i++)
    {
        cents = cents - 1;
    }
    return i;
}

File: sentimental-cash/cash.py
from cs50 import get_float

# Step 1 , get user input , with a check for it being a floating point postitive number
while True:
    cents = get_float("Change owed: ")
    if cents >= 0:
        break

cents = cents * 100  # convert dollar to cent equivalent

# Step 3 to calculate quarters, and remaining cents
quarters = int(cents/25)
cents = cents - quarters*25

# Repeat of step 3 for dimes
dimes = int(cents/10)
cents = cents - dimes*10

# Repeat of step 3 for nickels
nickels = int(cents/5)
cents = cents - nickels*5

# Repeat of step 3 for pennies
pennies = int(cents/1)
cents = cents - pennies*1

# Sum of all coins, which is the output
coins = quarters + dimes + nickels + pennies
print(coins)
